### PR TITLE
DYN-7671: pm client crash fix - version parsing

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

Fixes this issue: https://jira.autodesk.com/browse/DYN-7671

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [x] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB

### Release Notes

- bug when parsing short versions was causing the pm client to crash
- now safely pares versions
- moved method to VersionsUtility class
- test coverage

### Reviewers

@zeusongit 
@QilongTang 

### FYIs

@jnealb

### CODE REVIEW

File: src/DynamoPackages/PackageManagerSearchElement.cs [hunk 1]
```diff
@@ -352,8 +353,8 @@ private List<VersionInformation> TransformVersionsToVersionInformation(Greg.Resp
                 else
                 {
                     // Check within min and max range
-                    isWithinMinMax = (dynamoCompatibility.min == null || dynamoVersion >= Version.Parse(dynamoCompatibility.min)) &&
-                                     (dynamoCompatibility.max == null || dynamoVersion <= Version.Parse(dynamoCompatibility.max));
+                    isWithinMinMax = (dynamoCompatibility.min == null || dynamoVersion >= VersionUtilities.ParseVersionSafely(dynamoCompatibility.min)) &&
```
Comment (zeusongit) [inline src/DynamoPackages/PackageManagerSearchElement.cs:356]: Shouldn't `ParseVersionSafely` be the default `Parse` behavior?

File: src/DynamoPackages/PackageManagerSearchElement.cs [hunk 2]
```diff
@@ -395,7 +393,7 @@ private List<VersionInformation> TransformVersionsToVersionInformation(Greg.Resp
 
             // If we passed all the checks, the version is compatible
             return true;
-        }
+        }
```
Comment (zeusongit) [inline src/DynamoPackages/PackageManagerSearchElement.cs:396]: spaces

File: src/DynamoPackages/PackageManagerSearchElement.cs (hunk 1 above)
Comment (dnenov) [reply src/DynamoPackages/PackageManagerSearchElement.cs:356]: Sorry, what do you mean? The problem is that we are sending incomplete versions, like '2019' which fails. If we do 'Version.TryParse' we won't get the version out. So, we need a 'manual' way to get the correct version from an incomplete string version representation.

File: src/DynamoPackages/PackageManagerSearchElement.cs (hunk 2 above)
Comment (dnenov) [reply src/DynamoPackages/PackageManagerSearchElement.cs]: Yup, done!

File: src/DynamoPackages/PackageManagerSearchElement.cs (hunk 1 above)
Comment (zeusongit) [reply src/DynamoPackages/PackageManagerSearchElement.cs:356]: Sorry, I was not clear, by that I meant 2 things:
1. Why even parse "unsafely", this method should replace all instances of `Version.TryParse` or `Version.Parse`?
A few lines [above](https://github.com/DynamoDS/Dynamo/pull/15572/files#diff-a7975ce1cd3828b7bc9a238f4ae9e0e5352d4ab6fba2071899a423ffdf2286ffR305) I can see TryParse being used.
2. Why not name the method as just `VersionUtilities.Parse`?

Comment (dnenov) [reply src/DynamoPackages/PackageManagerSearchElement.cs:356]: Oh, I see! Good point, cover all the uses, of course. 

I will rename it as proposed, @zeusongit

**Review by zeusongit — APPROVED** [review]

**Pull request merged**